### PR TITLE
New error codes topic

### PR DESCRIPTION
**Pull request by hhunter-ms**
## Description

Create new error codes topic under Developing Applications with everyone's doc contributions moved to sit underneath it.
Add an 'overview' based on @alicejgibbons presentation

## Related

PR is a pt 2 of: #4422

### CODE REVIEW

**Comment by alicejgibbons** [comment]
Happy to review this @hhunter-ms TY for writing!

**Comment by hhunter-ms** [comment]
@alicejgibbons thanks Alice! lmk if there's anything I can edit to make it clearer.

File: daprdocs/content/en/developing-applications/error-codes/errors-overview.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+title: "Errors overview"
+linkTitle: "Overview"
+weight: 10
+description: "Overview of Dapr errors"
+---
+
+An error code is a numeric or alphamueric code that indicates the nature of an error and, when possible, why it occured. 
+
+Dapr error codes are standardized strings for over 80+ common errors across HTTP and gRPC requests when using the Dapr APIs. These codes are both:
+- Returned in the JSON response body of the request
```
Comment (msfussell) [inline daprdocs/content/en/developing-applications/error-codes/errors-overview.md]:
```suggestion
- Returned in the JSON response body of the request.
```

File: daprdocs/content/en/developing-applications/error-codes/errors-overview.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+The returned error includes:
+- The error code: `ERR_STATE_STORE_NOT_FOUND`
+- The error message describing the issue: `state store <name> is not found`
+- The app ID in which the error is occuring: `nodeapp`
+- The reason for the error: `DAPR_STATE_NOT_FOUND`
+
+## Dapr error code metrics
+
+Metrics help users see when exactly errors are occuring from within the runtime. Error code metrics are collected using the `error_code_total` endpoint. This endpoint is disabled by default. You can [enable it using the `recordErrorCodes` field in your configuration file]({{< ref "metrics-overview.md#configuring-metrics-for-error-codes" >}}).
```
Comment (msfussell) [inline daprdocs/content/en/developing-applications/error-codes/errors-overview.md]:
```suggestion
Metrics help you see when exactly errors are occuring from within the runtime. Error code metrics are collected using the `error_code_total` endpoint. This endpoint is disabled by default. You can [enable it using the `recordErrorCodes` field in your configuration file]({{< ref "metrics-overview.md#configuring-metrics-for-error-codes" >}}). 
```

File: daprdocs/content/en/developing-applications/error-codes/http-error-codes.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+---
+type: docs
+title: "Handling HTTP error codes"
+linkTitle: "HTTP"
+description: "Detailed reference of the Dapr HTTP error codes and how to handle them"
+weight: 30
+---
+
+For http calls made to Dapr runtime, when an error is encountered, an error json is returned in http response body. The json contains an error code and an descriptive error message, e.g.
```
Comment (msfussell) [inline daprdocs/content/en/developing-applications/error-codes/http-error-codes.md]:
```suggestion
For HTTP calls made to Dapr runtime, when an error is encountered, an error JSON is returned in response body. The JSON contains an error code and an descriptive error message. 
```

File: daprdocs/content/en/developing-applications/error-codes/http-error-codes.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+```
+{
+    "errorCode": "ERR_STATE_GET",
```
Comment (msfussell) [inline daprdocs/content/en/developing-applications/error-codes/http-error-codes.md:13]: Need to check, but I feel this error format is the same as the one above. List this. I need to check with Alice on this, or ask Jake or watch the video again

```json
{
  "error": "Bad Request",
  "error_msg": "{\"errorCode\":\"ERR_STATE_STORE_NOT_FOUND\",\"message\":\"state store <name> is not found\",\"details\":[{\"@type\":\"type.googleapis.com/google.rpc.ErrorInfo\",\"domain\":\"dapr.io\",\"metadata\":{\"appID\":\"nodeapp\"},\"reason\":\"DAPR_STATE_NOT_FOUND\"}]}",
  "status": 400

Comment (hhunter-ms) [reply daprdocs/content/en/developing-applications/error-codes/http-error-codes.md:13]: @msfussell what I have, I grabbed from the video while it was paused

**Review by msfussell — APPROVED** [review]
LGTM

**Pull request merged**